src/portfft/dispatcher/workitem_dispatcher.hpp

```diff
@@ -255,18 +305,43 @@ PORTFFT_INLINE void workitem_impl(const T* input, T* output, const T* input_imag
         }
       }
     }
-    if (LayoutOut == detail::layout::PACKED) {
+    // This is checking for LayoutOut is PACKED or UNPACKED but we don't actually ever launch kernels with LayoutOut
+    // as UNPACKED, but just use PACKED and check the stride.
+    if (LayoutOut == detail::layout::PACKED || LayoutOut == detail::layout::UNPACKED) {
       sycl::group_barrier(global_data.sg);
       global_data.log_dump_local("computed data local memory:", loc, n_reals * n_working);
       global_data.log_message_global(__func__, "storing data from local to global memory");
-      if (storage == complex_storage::INTERLEAVED_COMPLEX) {
-        local2global<level::SUBGROUP, SubgroupSize>(global_data, loc_view, output, n_reals * n_working, local_offset,
-                                                    global_offset);
+      if (output_stride == 1 && output_distance == fft_size) {
```

Comment:
That is because we never actually use instantiate LayoutOut == UNPACKED kernels, so its kind of check the it is actually packed. Its described in the comment directly above.
I can see why its confusing but a proper fix would be more disruptive.

Reply:
I think this relates to my other [comment](https://github.com/codeplaysoftware/portFFT/pull/136#discussion_r1478595269). I agree that this is a bit too confusing even if we expect this to be a temporary solution.